rails/app/models/api/v1/offering.rb

```diff
@@ -100,7 +100,13 @@ def initialize(offering, protocol, host_with_port, current_user, additional_exte
     end
     self.external_reports = []
     if runnable.respond_to?(:external_reports) && runnable.external_reports
-      runnable.external_reports.each do |report|
+      # Filter `external_reports` if `researcher_data` is present.
```

Comment:
I thought you would be filtering for this in the researcher class React component and not the backend.